src/server/editorServices.ts

```diff
@@ -1105,187 +1312,159 @@ namespace ts.server {
             }
         }
 
-        private reportConfigFileDiagnostics(configFileName: string, diagnostics: Diagnostic[], triggerFile: string) {
-            if (!this.eventHandler) {
-                return;
-            }
-
-            this.eventHandler(<ConfigFileDiagEvent>{
-                eventName: ConfigFileDiagEvent,
-                data: { configFileName, diagnostics: diagnostics || [], triggerFile }
-            });
-        }
-
-        private createAndAddConfiguredProject(configFileName: NormalizedPath, projectOptions: ProjectOptions, configFileErrors: Diagnostic[], clientFileName?: string) {
-            const sizeLimitExceeded = this.exceededTotalSizeLimitForNonTsFiles(configFileName, projectOptions.compilerOptions, projectOptions.files, fileNamePropertyReader);
+        private createAndAddConfiguredProject(configFileName: NormalizedPath, projectOptions: ProjectOptions, configFileErrors: Diagnostic[], configFileSpecs: ConfigFileSpecs, cachedServerHost: CachedServerHost, clientFileName?: string) {

```

Comment:
Its a fairly long function. I wouldn't recommend inlining unless the containing function is extremely short.